core/envelope-sim/src/main/java/fr/sncf/osrd/envelope_sim/allowances/AbstractAllowanceWithRanges.java

```diff
@@ -156,6 +156,9 @@ private Envelope[] computeAllowanceRegion(Envelope envelopeRegion, EnvelopeSimCo
         var rangePercentages = new RangePercentage[ranges.size()];
         for (var i = 0; i < ranges.size(); i++) {
             var range = ranges.get(i);
+            if (range.beginPos < envelopeRegion.getBeginPos() || range.endPos > envelopeRegion.getEndPos()) {
+                throw new OSRDError(ErrorType.AllowanceRangeOutOfBounds);
+            }
```

Comment:
I don't think this is the right place to do this.

There should be two assertions: one that the ranges are contained within the allowance bounds, and one that the allowance bounds are contained in the envelope. They should be done as soon as possible.

The first check should be done when initializing an `AbstractAllowanceWithRanges`. The constructor can call a new method that would check the inputs. 

The second one can't be checked without an input, so it should be done at the start of `apply`.

Reply:
Done.